OptimizelySDK/Optimizely.cs

```diff
@@ -338,54 +339,60 @@ private bool ValidateInputs(string datafile, bool skipJsonValidation)
         /// <summary>
         /// Sends conversion event to Optimizely.
         /// </summary>
-        /// <param name="eventKey">Event key representing the event which needs to be recorded</param>
+        /// <param name="eventKey">Event key representing the event (must not be null, empty, or whitespace)</param>
         /// <param name="userId">ID for user</param>
         /// <param name="userAttributes">Attributes of the user</param>
         /// <param name="eventTags">eventTags array Hash representing metadata associated with the event.</param>
         public void Track(string eventKey, string userId, UserAttributes userAttributes = null,
             EventTags eventTags = null
         )
         {
-            var config = ProjectConfigManager?.GetConfig();
-
-            if (config == null)
+            if (eventKey == null || Regex.IsMatch(eventKey, @"^\s*$"))
             {
-                Logger.Log(LogLevel.ERROR, "Datafile has invalid format. Failing 'Track'.");
+                Logger.Log(LogLevel.ERROR,
+                    "Event key cannot be null, empty, or whitespace string. Failing 'Track'.");
                 return;
             }
 
             var inputValues = new Dictionary<string, string>
-                { { USER_ID, userId }, { EVENT_KEY, eventKey } };
+            {
+                { USER_ID, userId },
+                { EVENT_KEY, eventKey },
+            };
 
             if (!ValidateStringInputs(inputValues))
             {
                 return;
             }
 
-            var eevent = config.GetEvent(eventKey);
-
-            if (eevent.Key == null)
```

Comment:
I think these 2 lines will filter out null or empty eventKey cases.

Reply:
I think I see what you're talking about: The eventKey must be in the datafile before it can be tracked anyway, regardless of whether it's not null, empty, or whitespace.